### PR TITLE
Add new meta

### PR DESCRIPTION
**Pull request by azoyan**

### CODE REVIEW

File: .metahash [hunk 1]
```diff
@@ -0,0 +1 @@
+[76, 70, 187, 76, 209, 187, 199, 103, 120, 33, 234, 63, 154, 62, 201, 86, 201, 150, 91, 206, 5, 174, 177, 21, 164, 93, 228, 199, 183, 111, 220, 135]
```
Comment (shamilsan) [inline .metahash]: We've agreed to remove `.metahash` from the repo.

File: meta.txt [hunk 2]
```diff
@@ -0,0 +1 @@
...75726174696f6e00000c0114686f7572731c010c75363400011c6d696e757465731c010c75363400011c7365636f6e64731c010c7536340000280c2861756374696f6e5f696f08696f144576656e7400010c3841756374696f6e537461727465640c012c746f6b656e5f6f776e657208011c4163746f724964000114707269636520011075313238000120746f6b656e5f69641401105532353600000018426f756768740401147072696365200110753132380001003441756374696f6e53746f70656408012c746f6b656e5f6f776e657208011c4163746f724964000120746f6b656e5f696414011055323536000200002c0c2861756374696f6e5f696f1c61756374696f6e1c41756374696f6e00001c01146f776e657208011c4163746f72496400010c6e667430010c4e46540001387374617274696e675f707269636520011075313238000134646973636f756e745f7261746520011075313238000118737461747573340118537461747573000128737461727465645f61741c010c753634000128657870697265735f61741c010c7536340000300c2861756374696f6e5f696f1c61756374696f6e0c4e465400000c0120746f6b656e5f6964140110553235360001146f776e657208011c4163746f72496400012c636f6e74726163745f696408011c4163746f7249640000340c2861756374696f6e5f696f1c61756374696f6e18537461747573000114104e6f6e6500000024497352756e6e696e67000100245075726368617365640401147072696365200110753132380002001c457870697265640003001c53746f7070656400040000
```
Comment (shamilsan) [inline meta.txt:1]: Have we agreed to keep `.metahash` and `meta.txt` in releases, aren't we?

File: io/src/auction.rs [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+    pub owner: ActorId,
+    pub nft: NFT,
+    pub starting_price: u128,
+    pub discount_rate: u128,
+    pub status: Status,
+    pub started_at: u64,
+    pub expires_at: u64,
+}
+
+impl Auction {
```
Comment (shamilsan) [inline io/src/auction.rs]: Usually, we hadn't `impl`s in `io` crate. Isn't it better to keep business logic in one crate?

File: src/contract.rs [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+extern "C" fn metahash() {
+    reply(include!("../.metahash"))
+        .expect("Failed to encode or reply with `[u8; 32]` from `metahash()`");
+}
+
+fn reply(payload: impl Encode) -> GstdResult<MessageId> {
+    msg::reply(payload, 0)
+}
+
+gstd::metadata! {
```
Comment (shamilsan) [inline src/contract.rs]: Old meta

File: tests/state_tests.rs [hunk 5]
```diff
@@ -8,10 +8,10 @@ use routines::*;
 // #[test]
 // fn is_not_active_after_time_is_over() {
 //     let sys = System::new();
-//
+
```
Comment (shamilsan) [inline tests/state_tests.rs]: 1. Why is this test disabled?
2. Why are there changes?

File: .metahash (hunk 1 above)
Comment (fluiderson) [reply .metahash]: This file is already in `.gitingore`. Just delete it from the repo.

File: Cargo.toml [hunk 6]
```diff
@@ -9,8 +9,9 @@ authors = ["Gear Technologies"]
 gstd = { git = "https://github.com/gear-tech/gear.git", features = ["debug"], branch = "stable" }
 auction-io = { path = "io" }
 primitive-types = { version = "0.12.1", default-features = false, features = ["scale-info"] }
-nft-io = { git = "https://github.com/gear-dapps/non-fungible-token.git", tag = "0.2.5" }
-gear-lib = { git = "https://github.com/gear-dapps/gear-lib.git", tag = "0.3.3" }
+nft-io = { git = "https://github.com/gear-dapps/non-fungible-token.git", branch = "new_meta" }
+gear-lib = { git = "https://github.com/gear-dapps/gear-lib.git", tag = "0.3.4" }
+gmeta = { git = "https://github.com/gear-tech/gear", features = ["codegen"] }
```
Comment (fluiderson) [inline Cargo.toml]:
```suggestion
gmeta = { git = "https://github.com/gear-tech/gear" }
```
Makes sense only in the `state`crate.

File: io/src/auction.rs [hunk 7]
```diff
@@ -0,0 +1,220 @@
+use crate::io::*;
+use core::cmp::min;
+use gstd::{exec, msg, prelude::*, ActorId};
+use nft_io::{NFTAction, NFTEvent};
+use primitive_types::U256;
+
+#[derive(Debug, Decode, Default, Encode, TypeInfo, Clone)]
+#[codec(crate = gstd::codec)]
+#[scale_info(crate = gstd::scale_info)]
```
Comment (fluiderson) [inline io/src/auction.rs]:
```suggestion
```
Unneeded since `io/Cargo.toml` contains the `scale-info` and `codec` deps.

File: io/src/auction.rs [hunk 8]
```diff
@@ -0,0 +1,220 @@
...
+#[derive(Debug, Clone, Default, Encode, Decode, TypeInfo)]
+pub struct NFT {
+    pub token_id: U256,
+    pub owner: ActorId,
+    pub contract_id: ActorId,
+}
+
+#[derive(Debug, Clone, Default, Encode, Decode, TypeInfo)]
+#[codec(crate = gstd::codec)]
+#[scale_info(crate = gstd::scale_info)]
```
Comment (fluiderson) [inline io/src/auction.rs]:
```suggestion
```
Unneeded since `io/Cargo.toml` contains the `scale-info` and `codec` deps.

File: io/src/auction.rs [hunk 9]
```diff
@@ -0,0 +1,220 @@
...
+            discount_rate: self.discount_rate,
+            time_left: self.expires_at.saturating_sub(exec::block_timestamp()),
+            status: self.status.clone(),
+        }
+    }
+}
+
+#[derive(Debug, Decode, Encode, TypeInfo)]
+#[codec(crate = gstd::codec)]
+#[scale_info(crate = gstd::scale_info)]
```
Comment (fluiderson) [inline io/src/auction.rs]:
```suggestion
```
Unneeded since `io/Cargo.toml` contains the `scale-info` and `codec` deps.

File: io/src/io.rs [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+    type Handle = InOut<Action, Event>;
+    type Others = ();
+    type Reply = ();
+    type Signal = ();
+    type State = Auction;
+}
+
+#[derive(Debug, Encode, Decode, TypeInfo)]
+#[codec(crate = gstd::codec)]
+#[scale_info(crate = gstd::scale_info)]
```
Comment (fluiderson) [inline io/src/io.rs]:
```suggestion
```
Unneeded since `io/Cargo.toml` contains the `scale-info` and `codec` deps.

File: io/src/io.rs [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+pub enum Action {
+    Buy,
+    Create(CreateConfig),
+    ForceStop,
+}
+
+#[derive(Debug, Encode, Decode, TypeInfo)]
+#[codec(crate = gstd::codec)]
+#[scale_info(crate = gstd::scale_info)]
```
Comment (fluiderson) [inline io/src/io.rs]:
```suggestion
```
Unneeded since `io/Cargo.toml` contains the `scale-info` and `codec` deps.

File: io/src/io.rs [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+    },
+    AuctionStoped {
+        token_owner: ActorId,
+        token_id: U256,
+    },
+}
+
+#[derive(Debug, Encode, Decode, TypeInfo)]
+#[codec(crate = gstd::codec)]
+#[scale_info(crate = gstd::scale_info)]
```
Comment (fluiderson) [inline io/src/io.rs]:
```suggestion
```
Unneeded since `io/Cargo.toml` contains the `scale-info` and `codec` deps.

File: io/src/io.rs [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+pub struct Duration {
+    pub hours: u64,
+    pub minutes: u64,
+    pub seconds: u64,
+}
+
+#[derive(Debug, Encode, Decode, TypeInfo)]
+#[codec(crate = gstd::codec)]
+#[scale_info(crate = gstd::scale_info)]
```
Comment (fluiderson) [inline io/src/io.rs]:
```suggestion
```
Unneeded since `io/Cargo.toml` contains the `scale-info` and `codec` deps.

**Review by LouiseMedova — APPROVED** [review]

File: tests/node_tests.rs [hunk 14]
```diff
@@ -1,18 +1,20 @@
...
-#[cfg(not(debug_assertions))]
-const PATH: &str = "../target/wasm32-unknown-unknown/release/dutch_auction.opt.wasm";
+const NFT_PATH: &str = "target/nft-0.2.5.opt.wasm";
 
 #[tokio::test]
-async fn buy() -> Result<()> {
+// #[ignore]
+async fn create_and_stop() -> Result<()> {
+    assert!(Path::new(NFT_PATH).exists());
+
```
Comment (fluiderson) [inline tests/node_tests.rs]: Why check? `gclient::code_from_os()` already do this internally.

File: tests/node_tests.rs [hunk 15]
```diff
@@ -1,18 +1,20 @@
...
-#[cfg(debug_assertions)]
-const PATH: &str = "../target/wasm32-unknown-unknown/debug/dutch_auction.opt.wasm";
-
-#[cfg(not(debug_assertions))]
-const PATH: &str = "../target/wasm32-unknown-unknown/release/dutch_auction.opt.wasm";
+const NFT_PATH: &str = "target/nft-0.2.5.opt.wasm";
 
 #[tokio::test]
-async fn buy() -> Result<()> {
+// #[ignore]
```
Comment (fluiderson) [inline tests/node_tests.rs]:
```suggestion
#[ignore]
```
Tests with `gclient` should be ignored.

**Review by fluiderson — APPROVED** [review]

**Pull request merged**